site/content/contribute/more-info/server/schema-migration-guide.md

```diff
@@ -22,15 +39,293 @@ A file naming convention is used to determine the order in which the migrations
+
+```
+CREATE INDEX - 489
+ALTER TABLE - 195
+  ADD COLUMN - 113
+  ALTER COLUMN - 51
+  DROP COLUMN - 25
+  ADD CONSTRAINT - 6
+DROP INDEX - 124
+CREATE TABLE - 60
+UPDATE - 19
+DELETE - 2
+```
+
+![Migrations distribution](/contribute/more-info/server/migrations_pie.png)
+
+We will go through each of these migration types and discuss how we can make it non-blocking. This is a lengthy document, so for those wanting to directly look at the executive summary, we present it right now. And then expand on each section in detail later.
+
+### Conclusions
+
+**MySQL 5.7**
+
+| Operation      | Table rewrite | Concurrent DML allowed |
+| -----------    | ------------- | ---------------------- |
+| CREATE INDEX   | NO<sup>1</sup>| YES (with limitations)<sup>1</sup>|
+| DROP INDEX     | NO            | YES                    |
+| ADD COLUMN     | YES           | YES<sup>2</sup>                   |
+| ALTER COLUMN   | YES           | NO                     |
+| DROP COLUMN    | YES           | YES                    |
+| ADD FK CONSTRAINT| NO          | NO (avoidable)<sup>3</sup>        |
+| ADD UNIQUE CONSTRAINT | NO     | YES                    |
+
+Note:
+1. Fulltext and spatial indexes take locks. Adding a fulltext index can also rewrite the table if there is no user-defined FTS_DOC_ID column, or in other words if a fulltext index is being added for the first time.
+2. Concurrent DML is also not permitted while adding an auto-increment column. However we don’t use it and are unlikely to use it in future. So I have made the cell green for now.
+3. Setting `foreign_key_checks` to false avoids table locks. But it also defeats the purpose of a foreign key.
+
+**MySQL 8**
+
+| Operation      | Table rewrite | Concurrent DML allowed |
+| -----------    | ------------- | ---------------------- |
+| CREATE INDEX   | NO<sup>1</sup>| YES (with limitations)<sup>1</sup>|
+| DROP INDEX     | NO            | YES                    |
+| ADD COLUMN     | NO (only in certain cases) <sup>2</sup>           | YES<sup>3</sup>                   |
+| ALTER COLUMN   | YES           | NO                     |
+| DROP COLUMN    | YES           | YES                    |
+| ADD FK CONSTRAINT| NO          | NO (avoidable)<sup>4</sup>        |
+| ADD UNIQUE CONSTRAINT | NO     | YES                    |
+
+Note:
+1. Fulltext and spatial indexes take locks. Adding a fulltext index can also rewrite the table if there is no user-defined FTS_DOC_ID column, or in other words if a fulltext index is being added for the first time.
+2. Table rewrite happens if the column is being added to a table with a fulltext index. Which means this will affect adding columns to the Posts or the Channels table for example.
+3. Concurrent DML is also not permitted while adding an auto-increment column. However we don’t use it and are unlikely to use it in future. So I have made the cell green for now.
```

Comment:
I think markdown wouldn't allow for colored tables?

Reply:
Yeah I looked very slightly into this. You _could_ technically do this in hugo with some custom css, but I didn't bother spending that much effort :stuck_out_tongue:

Reply:
If this is highly desirable, @emdecr may be able to assist.

Reply:
It will be good to have ;)

Reply:
@agnivade @cwarnermm Sorry I missed this - my notification routing was weird; didn't get pinged on my MM email 😅 

Anywho - Did y'all still want me to take a look at this? Just lemme know. Screencap attached. Was it this entire column that you wanted to colour green?

<img width="869" alt="Screen Shot 2023-05-16 at 10 37 29 AM" src="https://github.com/mattermost/mattermost-developer-documentation/assets/10912003/87f20cde-46d6-4771-b526-de4d2340ff22">

Thanks!

Reply:
@emdecr - Not the entire column, check this [doc](https://docs.google.com/document/d/1wdQLzS_C6cjJIcyZWEYEeaP3Ej_NuuXj4lDt1VFuyZU/edit#heading=h.xe5arfvo4667). Some cells will be green, red and yellow depending on their value. Basically we need to match whatever is there in the doc.

Let us know if it's doable :)

Reply:
No rush at all. Feel free to take your time!